src/routes/room.js

```diff
@@ -203,50 +226,32 @@ router.route('message/:sID').get(async(req, res) => {
 
 
 
-
 //upload a file to the database  file needs to be transformed to a buffer be being sent
 // the user should send he ID of the study room, 
-router.route('/file').post(async(req, res) => {
+router.post('/file', upload.single("file"), (req, res) => {
 
-    
-    let r = (Math.random() + 1).toString(36).substring(7);
+  let r = (Math.random() + 1).toString(36).substring(7);
+        const newImage= new courseNotes({
+         cnID:r,
+         sID:req.body.sID,
+         email:req.body.email,
+         filetype:req.body.type,
+         file:{
+          data: fs.readFileSync('uploads/' + req.file.filename),
```

Comment:
Modify this line to `data: req.body.file,` since I already pass the buffered and read version of the file. Also, please update the courseNotes array for the respective study room in the studyRooms collection for consistency. It is also required so I can request a list of the course notes and access them (I don't have any other way of getting all cnIDs for the study room from the backend).